src/3D/Camera.cpp

```diff
@@ -172,39 +228,98 @@ void Camera::handleMouseInput(const SDL_Event& e)
 	// Holding down the middle mouse button enables free look.
 	if (e.type == SDL_MOUSEMOTION && e.motion.state & SDL_BUTTON(SDL_BUTTON_MIDDLE))
 	{
-		glm::vec3 rot = GetRotation();
-
-		rot.y -= e.motion.xrel * _freeLookSensitivity * 0.1f;
-		rot.x -= e.motion.yrel * _freeLookSensitivity * 0.1f;
-
-		SetRotation(rot);
+		int width = 1920;
+		int height = 1080;
+		Game::instance()->GetWindow()->GetSize(width,height);
```

Comment:
Why reset the resolution? The resolution can be set from the command line if you don't like the current one.

Reply:
It's not resetting the resolution, I'm just initializing two variables to be immediately updated by the GetSize function, we need the width and height to calculate how much the yaw and pitch are dependent on the width and height.

Reply:
Ah yes. You don't need to give these values. It can become confusing if you do that.

Reply:
did not change to vec2, just removed unnecessary initialization